### ISSUE TITLE
Metrics do not support multilabel tasks.

### ISSUE THREAD

**Issue by jdavid04** [PyTorchLightning/pytorch-lightning#4238]
## 🐛 Bug

Scikit-learn metrics deal well will multilabel tasks, but this doesn't seem to be supported in Pytorch-Lightning metrics.  There is this #3350 , but it seems to confuse multiclass with multilabel (multiple values to predict). 

### To Reproduce
Given predictions tensor: 
```
tensor([[0., 0.],
              [0., 0.],
              [0., 0.],
              [0., 0.],
              [0., 0.],
              [0., 0.],
              [0., 0.],
              [0., 0.]])
```
and labels tensor:
 ```
tensor([[1, 0],
              [1, 0],
              [1, 0],
              [1, 0],
              [1, 0],
              [1, 0],
              [1, 0],
              [1, 0]])

```
The call to `f1_score(met_preds,labels, class_reduction='macro')` yields `tensor(0.3333)`, because it flattens the tensors and macro-averages per class.

### Expected behavior

I would expect it to be consistent with the call to
`sk_f1_score(labels.numpy(), met_preds.numpy(), average='macro')`, which yields `0.0`, because it treats each column separately and macro-averages them per task.

This discrepancy also occurs for other metrics. For example sklearn deals with multilabel accuracy by using subset accuracy (0 here), but PL produces an accuracy score of 0.5.

### Environment
 - PyTorch Version : 1.6
 - OS (e.g., Linux): OSX
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Python version: 3.7.8
- Pytorch-Lightning version : 1.0.2

**Comment by Borda** [comment]
@justusschock mind have look? :]

**Comment by edenlightning** [comment]
@teddykoker mind taking a look?

**Comment by teddykoker** [comment]
Could you try the class interface? We have tested multilabel f1 with the class metrics, I think we are in the process of making sure we have the same functionality for functional

**Comment by SkafteNicki** [comment]
As @teddykoker stated we are in the process of unifying the metrics class based metrics with the functional metrics (starting with all the regression metrics here #4166 ). Until then, please use the `Fbeta` metric (however, note that there is currently a bug in that metric, that should be taken care of by this PR #4183 ):
```
metric = Fbeta(beta=1.0, multilabel=True, average='macro)
metric(preds, target)
>>> tensor(0.)
```

**Comment by jdavid04** [comment]
Thanks! Will stick to class-based metrics for the time being.

**Comment by edenlightning** [comment]
@teddykoker @SkafteNicki can this be closed? is this fixed by #4166?

**Comment by SkafteNicki** [comment]
Yes, let's close it. I will unify fbeta functional and class metric within a couple of days, so functional also support multilabel.
